source/hic_hal/stm32/stm32h743ii/extended_features/include/adapter_detector.h

```diff
@@ -0,0 +1,21 @@
+#ifndef _ADAPTER_DETECTOR_H_INCLUDED_
+#define _ADAPTER_DETECTOR_H_INCLUDED_
+
+#include <stdint.h>
+
+typedef enum
+{
+    ADAPTER_UDB_13_FLEX,
+    ADAPTER_JOKER_FLEX,
+    ADAPTER_UDB_6_FLEX,
+    ADAPTER_UDB_17_FLEX,
+    ADAPTER_UDB_12_FLEX,
+    ADAPTER_UNKNOWN,
```

Comment:
I suggest reserving a number outside of the range of normal types for the UNKNOWN and NONE types. You could maybe have ADAPTER_NONE = 0 and ADAPTER_UNKNOWN = 0xFF or something along those lines (not sure what value is most suitable here). This is because at some point you will add more adapter types, and then the numbers that used to mean UNKNOWN and NONE will mean something else. Better to keep them the same, especially if these types will be part of the software interface at some point.